lib/internal/fs/streams.js

```diff
@@ -111,23 +112,34 @@ function ReadStream(path, options) {
 Object.setPrototypeOf(ReadStream.prototype, Readable.prototype);
 Object.setPrototypeOf(ReadStream, Readable);
 
-ReadStream.prototype.open = function() {
-  fs.open(this.path, this.flags, this.mode, (er, fd) => {
+const openReadFs = internalUtil.deprecate(function() {
+  _openReadFs(this);
+}, 'ReadStream.prototype.open() is deprecated', 'DEP0XXX');
+ReadStream.prototype.open = openReadFs;
+
+function _openReadFs(stream) {
+  // Backwards compat for overriden open.
+  if (stream.open !== openReadFs) {
+    stream.open();
+    return;
+  }
```

Comment:
This broke the `fs-capacitor` module (used by `graphql-upload` to handle file uploads).
I don't know if that module could be refactored to fix this but currently it overloads the `open` method (and still calls it internally) so this results in an infinite recursion: https://github.com/mike-marcacci/fs-capacitor/blob/51ce05dfa8bfa7d915447d700d36e422480c43e0/src/index.mjs#L70-L79

Reply:
/cc @ronag @mcollina

Reply:
Can you open a new issue for this?